micro-services/school/src/main/java/com/exalt/training/school/SchoolService.java

```diff
@@ -0,0 +1,38 @@
+package com.exalt.training.school;
+
+import com.exalt.training.school.client.StudentClient;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+@RequiredArgsConstructor
+public class SchoolService {
+
+    private final SchoolRepository schoolRepository;
+    private final StudentClient client;
+
+
+    public void saveSchool(School school) {
+        schoolRepository.save(school);
+    }
+
+    public List<School> finalAllSchools() {
+        return schoolRepository.findAll();
+    }
+
+    public FullSchoolResponse findSchoolsWithStudents(Integer schoolId) {
+        var school = schoolRepository.findById(schoolId).orElse(School.builder()
```

Comment:
why you assignee student to not-found school!
each student should belong to one school at least, and student can move from one school to other schools. 
in the student profile, you should have an active school and history for schools